### DISCUSSION TITLE
Allow `item_forms` to have a `title` and `template`

### DISCUSSION THREAD

**Discussion by kiancross** [pretix/pretix#3480] [Ideas] [answered]
Currently, I don't think fields in `item_forms` can be grouped together, e.g. with a title/description.

On the other hand, `item_formsets` are far more customisable. 

What do you think about having an optional `title` and `template` field for `item_forms`, which behave in the same way as they do for `item_formsets`? Backwards compatibility would be maintained.

Happy to work on this if you think you'd accept a PR.

**Comment by raphaelm** [comment, marked as answer]
Yes, I believe I'd be interested in such a PR!

**Reply by kiancross** [reply]
Are there any examples of tests that test functionality provided by signals? e.g., I want to write a test which first returns a form on the `item_form` signal and then visits the item edit page to ensure the form is visible. I can see how to do the latter from existing tests, but less sure how to set up the signal.

**Reply by raphaelm** [reply]
We usually do not have good test coverage around these boundaries. When we do, we use the "fake plugin" that lives in tests/testdummy and make it listen to the signal
